main.go

```diff
@@ -34,7 +34,9 @@ func init() {
 var (
 	listenAddress       = flag.String("web.listen-address", ":9102", "The address on which to expose the web interface and generated Prometheus metrics.")
 	metricsEndpoint     = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
-	statsdListenAddress = flag.String("statsd.listen-address", ":9125", "The UDP address on which to receive statsd metric lines.")
+	statsdListenAddress = flag.String("statsd.listen-address", ":9125", "The UDP/TCP address on which to receive statsd metric lines.")
+	statsdListenUDP     = flag.Bool("statsd.listen-udp", true, "Whether to receive UDP statsd metrics.")
```

Comment:
doing it like what you mentioned have two conerns for me:
1. a bit uncomfortable for setting something to "" to disable it; and since we're default both UDP/TCP to on, use have to explicitly disable it;
2. the flags changes will break the user interface, from *statsdListenAddress* to *statsdListenUDP*. 

i have no strong feeling for 1; and for 2, if that's OK, i will do the flags changes.